scripts/update_lock.py

```diff
@@ -63,19 +63,16 @@ def fix_temp_dir_comments(file_path: Path):
 def is_file_outdated(old_file: Path, new_file: Path) -> bool:
     """Compares two lock files and returns True if there is a difference, else False"""
 
-    header_comment = "#    pip-compile"
     outdated = False
 
     with open(old_file, encoding="utf-8") as old:
         with open(new_file, encoding="utf-8") as new:
-            old_lines = old.readlines()
-            new_lines = new.readlines()
+            old_lines = old.readlines()[2:]  # skip header lines
+            new_lines = new.readlines()[2:]
```

Comment:
How about this, which ignores white space and comment lines and does not read the whole file, just until the first difference, thanks to the power of iterators:

from itertools import zip_longest

```python
def is_file_outdated(old_file: Path, new_file: Path) -> bool:
    """Compares two lock files and returns True if there is a difference, else False"""

    with open(old_file, encoding="utf-8") as old:
        with open(new_file, encoding="utf-8") as new:
            outdated = any(old_line != new_line for old_line, new_line in zip_longest(
                (line for line in (line.strip() for line in old)
                if line and not line.startswith('#')),
                (line for line in (line.strip() for line in new)
                if line and not line.startswith('#'))))

    if outdated:
        cli.echo_failure(f"{str(old_file)} is out of date!")
    return outdated
```

Reply:
We need zip_longest instead of zip to detect the case of one file being longer.